src/Microsoft.Azure.WebJobs.Host/Bindings/SystemBindingData.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Host.Bindings.Path;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    /// <summary>
+    /// Built in object for all binding expressions. 
+    /// </summary>
+    /// <remarks>
+    /// It's expected this class is created and added to the binding data. 
+    /// </remarks>
+    internal class SystemBindingData
+    {
+        // The public name for this binding in the binding expressions. 
+        public const string Name = "sys";
+
+        // An internal name for this binding that uses characters that gaurantee it can't be overwritten by a user. 
+        // This ensures that we can always unambiguously retrieve this later. 
+        private const string InternalKeyName = "$sys";
```

Comment:
It is a correctness issue. I need a key that is guaranteed to not conflict with user values; so the $ character gives me that.  But a user could say "ms_prefix". 
Request property bag doesn't have this issue since all access is internal. But Binding data gets populated with public values. 
Ultimately, it shouldn't matter - this is a private name, only used in this class, not visible to the user, and not persisted. So it could be changed later (or even completely removed from the binding data if we came up with another means). 

---
In reply to: [117296062](https://github.com/Azure/azure-webjobs-sdk/pull/1148#discussion_r117296062) [](ancestors = 117296062)